### PR TITLE
chore: migrating radiotab

### PR DESCRIPTION
**Pull request by quaresc**
## DESCRIPTION

Migrating radiotab
⚠️ The only difference I see is how it's displayed but I wonder if it's because of the old RadioGroup? If I remove `flex: 1` it's good but I wonder if that's what we want though

⚠️ It looks alright but I didn't necessarily re-implemented things from `${defaultFieldStyles({ size, variant })};` I don't know if I'm missing something or not :(

## HOW TO TEST

http://localhost:3020/components/radio-tab

## SCREENSHOTS / SCREEN RECORDINGS

https://github.com/user-attachments/assets/4d2f9fca-4c17-4da2-ab23-a31f3cee8356

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [ ] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [ ] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/RadioTab/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { Button } from '@/components/Button'
+import { classNames } from '@/utils'
+
+import styles from './radio-tab.module.scss'
+import type { RadioTabsProps } from './types'
+
+const cx = classNames(styles)
+
+export const RadioTab = forwardRef<HTMLInputElement, RadioTabsProps>(
+  ({ checked, dataTestId, disabled, id, label, onChange, value }, ref) => {
```
Comment (theo-mesnil) [inline lib/src/components/RadioTab/index.tsx]:
```suggestion
  ({ checked, dataTestId, disabled, id, label, onChange, value, className, ...rest }, ref) => {
```

File: lib/src/components/RadioTab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+      <Button
+        as="label"
+        className={cx('label', { checked })}
```
Comment (theo-mesnil) [inline lib/src/components/RadioTab/index.tsx]:
```suggestion
        className={cx('label', { checked }, className)}
```

File: lib/src/components/RadioTab/index.tsx (hunk 1 above)
Comment (theo-mesnil) [inline lib/src/components/RadioTab/index.tsx]: We want to remove dataTestId, it will be directly on rest

File: lib/src/components/RadioTab/index.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        data-testid={dataTestId}
+        disabled={disabled}
+        variant={checked ? 'primary' : null}
```
Comment (theo-mesnil) [inline lib/src/components/RadioTab/index.tsx:20]:
```suggestion
        disabled={disabled}
        variant={checked ? 'primary' : null}
        {...rest}
```

File: lib/src/components/RadioTab/radio-tab.module.scss [hunk 4]
```diff
@@ -0,0 +1,97 @@
+@use './../../utils/text-ellipsis' as ellipsis;
+
+@layer components {
+  .radio {
```
Comment (theo-mesnil) [inline lib/src/components/RadioTab/radio-tab.module.scss]:
```suggestion
  .root {
```

File: lib/src/components/RadioTab/types.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  size: Size
+  transparent?: boolean
+  value: Ariakit.RadioStoreState['value']
+  variant: 'danger' | 'success' | 'warning'
+}
+
+export type RadioTabsProps = ComponentPropsWithRef<'input'> & RadioTabsOptions
+
+export type Size = 'lg' | 'md' | 'sm' | 'xl' | 'xxl'
```
Comment (theo-mesnil) [inline lib/src/components/RadioTab/types.ts]: Do not export and put it directly on RadioTabsOptions

File: lib/src/components/RadioTab/radio-tab.module.scss [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    position: absolute;
+    top: 0;
+    left: 0;
+    opacity: 0;
+  }
+
+  .label {
+    flex: 1;
+    display: flex;
```
Comment (cnairi) [inline lib/src/components/RadioTab/radio-tab.module.scss]:
```suggestion
      display: inline-flex;
```
should fix your display issue

File: lib/src/components/RadioTab/index.tsx (hunk 1 above)
Comment (Isaacmaamouche) [reply lib/src/components/RadioTab/index.tsx]: Attention: leaving `dataTestId `can result in a unrecognized data attribute. 
Valid html attribute has to be data-testid, so Clément's implement is correct ❤️ 
```
        data-testid={dataTestId}

```

File: lib/src/components/RadioTab/docs/examples/disabled.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+export const ITEMS = [
+  { label: 'Bold', value: 'bold' },
+  { label: 'Italic', value: 'italic' },
+  { label: 'Strikethrough', value: 'strikethrough' },
+  { label: 'Underline', value: 'underline' },
+]
+
+const Example = () => {
+  return <RadioGroup disabled name="social" options={ITEMS.slice(0, 5)} renderOption={RadioTab} />
```
Comment (Isaacmaamouche) [inline lib/src/components/RadioTab/docs/examples/disabled.tsx]: Not related to your work but `ITEMS.slice(0, 5)` does nothing lmao

File: lib/src/components/RadioTab/index.tsx (hunk 1 above)
Comment (cnairi) [reply lib/src/components/RadioTab/index.tsx]: I kept it for this one + we need it for the Radio inpute:  data-testid={dataTestId ? `${dataTestId}-input` : undefined}

File: lib/src/components/RadioTab/index.tsx (hunk 3 above)
Comment (cnairi) [reply lib/src/components/RadioTab/index.tsx:20]: I kept the dataTestId as we need it for this one ( data-testid={dataTestId ? `${dataTestId}-input` : undefined})

File: lib/src/components/RadioTab/radio-tab.module.scss (hunk 4 above)
Comment (cnairi) [reply lib/src/components/RadioTab/radio-tab.module.scss]: It wasn't really a root, as it was applied to the Radio component, itself nested in a Button.
I replaced the current "input" class with "input-wrapper" class, and then the radio was just "input-content"

It's looks quite logical to me when inspecting the DOM:
Capture d’écran 2025-09-25 à 14 12 47

Let me know if you disagree

File: lib/src/components/RadioTab/types.ts (hunk 5 above)
Comment (cnairi) [reply lib/src/components/RadioTab/types.ts]: I completely removed it, Sizes were coming from the Button props

File: lib/src/components/RadioTab/docs/examples/disabled.tsx (hunk 7 above)
Comment (cnairi) [reply lib/src/components/RadioTab/docs/examples/disabled.tsx]: you're right, I removed itfor both examples

**Review by Leiksa — APPROVED** [review]

**Review by cnairi — APPROVED** [review]
